sdk/src/test/scala/com/horizen/SidechainMemoryPoolTest.scala

```diff
@@ -0,0 +1,222 @@
+package com.horizen
+
+import java.util
+
+import org.scalatestplus.junit.JUnitSuite
+import org.junit.{Before, Test}
+import org.junit.Assert._
+import com.horizen.fixtures._
+import com.horizen.transaction.RegularTransaction
+import com.horizen.utils.FeeRate
+import org.mockito.Mockito
+import scorex.util.ModifierId
+import org.scalatestplus.mockito.MockitoSugar
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+class SidechainMemoryPoolTest
+  extends JUnitSuite
+  with TransactionFixture
+  with SidechainTypesTestsExtension
+  with MockitoSugar
+{
+
+  @Test
+  def remove(): Unit = {
+    val memoryPool = SidechainMemoryPool.createEmptyMempool(getMockedMempoolSettings(300, 0))
+    val tx = getRegularTransaction
+    var tx2 = getCompatibleTransaction
+    val txId: ModifierId = ModifierId @@ tx.id
+    assertEquals("maxPoolSizeBytes must be equals to 300MB size",
+      memoryPool.maxPoolSizeBytes, (300*1024*1024))
+
+    assertEquals("Put operation must be success.", memoryPool.put(tx).isSuccess, true)
+    assertEquals("Size must be 1.", memoryPool.size, 1)
+    assertEquals("MemoryPool must contain transaction" + txId, memoryPool.modifierById(txId).get, tx)
+
+    assertEquals("Put operation must be success.", memoryPool.put(tx2).isSuccess,
+      true)
+    assertEquals("Size must be 2.", memoryPool.size, 2)
+    assertEquals("usedPoolSizeBytes must be equals to tx1+tx2 size", memoryPool.usedPoolSizeBytes,
+      tx.size() + tx2.size())
+
+    memoryPool.remove(tx)
+    assertEquals("Size must be 1.", memoryPool.size, 1)
+    assertEquals("usedPoolSizeBytes must be equals to tx2 size", memoryPool.usedPoolSizeBytes, tx2.size())
+
```

Comment:
add the test case where trying to remove nonexisting Tx, check resulting usedPoolSizeBytes